app/controllers/carts_controller.rb

```diff
@@ -7,4 +8,19 @@ def show
     @show_comments = true
   end
 
+  def index
+    @role = 'requester'
+    my_carts = Cart.joins(:approvals).where(:approvals => {:role => @role, :user_id => @current_user[:id]})
+    if my_carts.empty?
+      @role = 'approver'
+      my_carts = Cart.joins(:approvals).where(:approvals => {:role => @role, :user_id => @current_user[:id]})
+    end
+    @closed_carts = my_carts.closed
+    @open_carts = my_carts.open
+  end
+
+  def archive
+    @role = params[:role] || 'requester'
+    @closed_cart_full_list = Cart.joins(:approvals).where(:approvals => {:role => @role, :user_id => @current_user[:id]}).closed
```

Comment:
oooh, nice!

On Fri, Jan 2, 2015 at 12:24 PM, Aidan Feldman <notifications@github.com>
wrote:

> In app/controllers/carts_controller.rb
> <https://github.com/18F/C2/pull/131#discussion-diff-22421660>:
>
> > @@ -7,4 +8,19 @@ def show
> >      @show_comments = true
> >    end
> >
> > +  def index
> > +    @role = 'requester'
> > +    my_carts = Cart.joins(:approvals).where(:approvals => {:role => @role, :user_id => @current_user[:id]})
> > +    if my_carts.empty?
> > +      @role = 'approver'
> > +      my_carts = Cart.joins(:approvals).where(:approvals => {:role => @role, :user_id => @current_user[:id]})
> > +    end
> > +    @closed_carts = my_carts.closed
> > +    @open_carts = my_carts.open
> > +  end
> > +
> > +  def archive
> > +    @role = params[:role] || 'requester'
> > +    @closed_cart_full_list = Cart.joins(:approvals).where(:approvals => {:role => @role, :user_id => @current_user[:id]}).closed
>
> This can be simplified to
>
> @closed_cart_full_list = current_user.carts.where(approvals: {role: @role}).closed
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/18F/C2/pull/131/files#r22421660>.
>